I get this error each time I logon

**Describe the bug**
An error is thrown each time i logon to my server

**To Reproduce**
Steps to reproduce the behavior:
1. Logon

**Expected behavior**
A clear and concise description of what you expected to happen.

**Plugin version**
- [X ] Using the latest released version of the plugin (Version 2.2.2)
- [ ] Using the `legacy` jar

**Server version**
Spigot 1.13.2

**Additional context and logs**
Add any other context about the problem here. Also post error messages and logs (use a service like Hastebin for logs, please!) in this area.

```
[12:06:31 ERROR]: Could not pass event PlayerSpawnLocationEvent to PerWorldInventory v2.2.2
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:309) ~[spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.PlayerList.a(PlayerList.java:122) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.LoginListener.b(LoginListener.java:149) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.LoginListener.tick(LoginListener.java:53) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.NetworkManager.a(NetworkManager.java:230) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.ServerConnection.c(ServerConnection.java:119) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:993) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:830) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:728) [spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.lang.ClassCastException: java.lang.Float cannot be cast to java.lang.Double
	at me.ebonjaeger.perworldinventory.serialization.LocationSerializer.deserialize(LocationSerializer.kt:38) ~[?:?]
	at me.ebonjaeger.perworldinventory.data.FlatFile.getLogout(FlatFile.kt:148) ~[?:?]
	at me.ebonjaeger.perworldinventory.listener.player.PlayerSpawnLocationListener.onPlayerSpawn(PlayerSpawnLocationListener.kt:31) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_171]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_171]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_171]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_171]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:305) ~[spigot-1.13.2.jar:git-Spigot-518206a-a5b9c7b]
	... 13 more

```

--- a/src/main/kotlin/me/ebonjaeger/perworldinventory/serialization/LocationSerializer.kt
+++ b/src/main/kotlin/me/ebonjaeger/perworldinventory/serialization/LocationSerializer.kt
@@ -17,9 +17,9 @@ object LocationSerializer
     {
         val obj = JSONObject()
         obj["world"] = location.world.name
-        obj["x"] = location.x
-        obj["y"] = location.y
-        obj["z"] = location.z
+        obj["x"] = location.x.toFloat()
+        obj["y"] = location.y.toFloat()
+        obj["z"] = location.z.toFloat()
         obj["pitch"] = location.pitch
         obj["yaw"] = location.yaw
 
@@ -35,9 +35,9 @@ object LocationSerializer
     fun deserialize(obj: JSONObject): Location
     {
         val world = Bukkit.getWorld(obj["world"] as String)
-        val x = obj["x"] as Double
-        val y = obj["y"] as Double
-        val z = obj["z"] as Double
+        val x = obj["x"] as Float
+        val y = obj["y"] as Float
+        val z = obj["z"] as Float
         val pitch = obj["pitch"] as Float
         val yaw = obj["yaw"] as Float
 

--- a/src/test/kotlin/me/ebonjaeger/perworldinventory/serialization/LocationSerializerTest.kt
+++ b/src/test/kotlin/me/ebonjaeger/perworldinventory/serialization/LocationSerializerTest.kt
@@ -46,9 +46,9 @@ class LocationSerializerTest
     private fun assertHasSameProperties(given: Location, expected: Location)
     {
         assertThat(given.world.name, equalTo(expected.world.name))
-        assertThat(given.x, equalTo(expected.x))
-        assertThat(given.y, equalTo(expected.y))
-        assertThat(given.z, equalTo(expected.z))
+        assertThat(given.x.toFloat(), equalTo(expected.x.toFloat()))
+        assertThat(given.y.toFloat(), equalTo(expected.y.toFloat()))
+        assertThat(given.z.toFloat(), equalTo(expected.z.toFloat()))
         assertThat(given.pitch, equalTo(expected.pitch))
         assertThat(given.yaw, equalTo(expected.yaw))
     }

